src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala

```diff
@@ -424,18 +428,27 @@ final case class ProcessApiServiceImpl(
       latestPublishedOnly
     )
     val result = for {
-      statusEnum <- state.traverse(CatalogManagementDependency.EServiceDescriptorState.fromValue).toFuture
-      callerSubscribedEservices <- agreementManagementService.getAgreementsByConsumerId(callerId)
-      agreementStates           <- parseArrayParameters(agreementState).traverse(AgreementState.fromValue).toFuture
-      retrievedEservices        <- retrieveEservices(producerId, consumerId, statusEnum, agreementStates)
-      eservices                 <- processEservicesWithLatestFilter(retrievedEservices, latestPublishedOnly)
-      eserviceAndSelfcareId     <- Future.traverse(eservices.toList)(getGetEserviceAndSelfcareId)
-      organizationsDetails <- partyManagementService.getBulkInstitutions(eserviceAndSelfcareId.map(_._2.toString()))
+      statusEnum            <- state.traverse(CatalogManagementDependency.EServiceDescriptorState.fromValue).toFuture
+      agreementStates       <- parseArrayParameters(agreementState).distinct.traverse(AgreementState.fromValue).toFuture
+      retrievedEservices    <- retrieveEservices(producerId, consumerId, statusEnum, agreementStates)
+      eservices             <- processEservicesWithLatestFilter(retrievedEservices, latestPublishedOnly)
+      eserviceAndSelfcareId <- Future.traverse(eservices.toList)(getGetEserviceAndSelfcareId)
+      agreements            <- Future
```

Comment:
this updated call improves performances compared to the previous one that was taking all consumer agreements.
It's still not perfect, because in `retrieveEservices` another `getAgreements` is performed.
Due to time constraint, I didn't spend time on refactor. (anyway this will be moved to BFF)